### PR TITLE
Fix more deprecations and also support Julia v0.7

### PR DESCRIPTION
**Pull request by rdeits**
This is a PR against your branch from https://github.com/vtjnash/Pidfile.jl/pull/2 (so merging it will just update that PR)

### CODE REVIEW

File: src/Pidfile.jl [hunk 1]
```diff
@@ -208,7 +208,7 @@ function open_exclusive(path::String;
         # now try again to create it
         file = tryopen_exclusive(path, mode)
         file === nothing || return file
-        wait(t) # sleep for a bit before trying again
+        fetch(t) # sleep for a bit before trying again
```
Comment (ranjanan) [inline src/Pidfile.jl]: Jameson didn't want this fixed: 
https://github.com/vtjnash/Pidfile.jl/pull/2#discussion_r209424875

File: test/runtests.jl [hunk 2]
```diff
@@ -135,7 +135,7 @@ end
         close(f2)
     end
     rm("pidfile")
-    wait(rmtask)
+    fetch(rmtask)
```
Comment (ranjanan) [inline test/runtests.jl]: Same here (
https://github.com/vtjnash/Pidfile.jl/pull/2#discussion_r209424875)

File: test/runtests.jl [hunk 3]
```diff
@@ -165,7 +165,7 @@ end
...
```
Comment (ranjanan) [inline test/runtests.jl]: https://github.com/vtjnash/Pidfile.jl/pull/2#discussion_r209424875

File: src/Pidfile.jl (hunk 1 above)
Comment (rdeits) [reply src/Pidfile.jl]: Oh, yikes, ok. I'll update this PR

**Comment by rdeits** [comment]
done

**Pull request merged**